distutils/compilers/C/msvc.py

```diff
@@ -371,17 +371,22 @@ def out_extensions(self) -> dict[str, str]:
 
     def compile(  # noqa: C901
         self,
-        sources,
+        sources: Sequence[str | os.PathLike[str]],
         output_dir=None,
         macros=None,
         include_dirs=None,
         debug=False,
         extra_preargs=None,
         extra_postargs=None,
         depends=None,
-    ):
+    ) -> list[str]:
         if not self.initialized:
             self.initialize()
+
+        # Move .mc files to the start of the list, otherwise keep the same order
+        # See pypa/setuptools#4986
+        sources = sorted(sources, key=lambda source: not str(source).endswith(".mc"))
```

Comment:
I could also look for the extensions in `_mc_extensions` instead of "hardocding" `.mc`.